From 88151d1e26b68580c0c1ea23469b069334678b36 Mon Sep 17 00:00:00 2001
From: Cheyenne Wills <cwills@sinenomine.net>
Date: Thu, 15 Aug 2024 16:08:18 -0600
Subject: [PATCH 4/4] rx: Remove includes for MIN/MAX in rx_packet.h

The commit:
    "opr: replace MIN/MAX macros with opr_min/opr_max"
       (Change I2d7b54193ec91f7ead9c5c5f714d9a8bc7533bf7)
replaced all uses of the MIN and MAX macros with opr_min and opr_max.

The include for sys/param.h and sys/sysmacros.h in rx_packet.h are no
longer needed for MIN/MAX.

Remove the preprocessor conditionals and the includes for param.h and
sysmacros.h.

Reviewed-on: https://gerrit.openafs.org/15823
Reviewed-by: Andrew Deason <adeason@sinenomine.net>
Reviewed-by: Michael Meffie <mmeffie@sinenomine.net>
Tested-by: BuildBot <buildbot@rampaginggeek.com>
(cherry picked from commit d299dfbfe90482b26b5f5551eb5be1b8e76d6ac4)

Change-Id: Ic1dec6f69bb8f9fd6eaffd0d893f0b2113193ae7
---
 src/rx/rx_packet.h | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/src/rx/rx_packet.h b/src/rx/rx_packet.h
index 48736d3f8..173762af7 100644
--- a/src/rx/rx_packet.h
+++ b/src/rx/rx_packet.h
@@ -31,20 +31,6 @@
  * and that the offsets are as well.
  */
 
-
-#if !defined(AFS_NT40_ENV)
-# if !defined(AFS_DARWIN_ENV) && !defined(AFS_USR_DARWIN_ENV)   \
-    && !defined(AFS_XBSD_ENV) && !defined(AFS_USR_FBSD_ENV)     \
-    && !defined(AFS_USR_DFBSD_ENV) && !defined(AFS_LINUX_ENV)
-#  include <sys/sysmacros.h>	/* MIN, MAX on most commercial UNIX */
-# endif
-/* Linux 3.7 doesn't have sys/param.h in kernel space, and afs/param.h ensures
- * that MIN and MAX are available for kernel builds. */
-# if !(defined(AFS_LINUX_ENV) && defined(KERNEL))
-#  include <sys/param.h>	/* MIN, MAX elsewhere */
-# endif
-#endif /* !AFS_NT40_ENV */
-
 #define	IPv6_HDR_SIZE		40	/* IPv6 Header */
 #define IPv6_FRAG_HDR_SIZE	 8	/* IPv6 Fragment Header */
 #define UDP_HDR_SIZE             8	/* UDP Header */
-- 
2.46.1

